mods/binoculars/init.lua

```diff
@@ -11,32 +17,49 @@ local creative_mode_cache = minetest.settings:get_bool("creative_mode")
 -- Update player property
 -- Global to allow overriding
 
-function binoculars.update_player_property(player)
+function binoculars.update_player_property(player, join)
+	local player_name = player:get_player_name()
 	local creative_enabled =
-		(creative and creative.is_enabled_for(player:get_player_name())) or
+		(creative_mod and creative.is_enabled_for(player_name)) or
 		creative_mode_cache
 
-	if creative_enabled or
-			player:get_inventory():contains_item("main", "binoculars:binoculars") then
```

Comment:
Let me simplify that for you:
```Lua

local zoom = creative_enabled or
	player:get_inventory():contains_item("main", "binoculars:binoculars")
if player:get_properties().can_zoom ~= zoom then
	player:set_properties({can_zoom = zoom})
end
```

Reply:
Ugh of course, i don't need the table at all, thanks.